pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -1,134 +1,163 @@
-		assert.Equal(t, username, newUser.Name)
-	})
+	// Objects to track in the fake client.
+	objs := []runtime.Object{userAcc}
 
-	t.Run("get_user", func(t *testing.T) {
-		userAcc := newUserAccount(username, userID)
-		user := newUser(userAcc)
-		client := fake.NewFakeClient(user)
-		reconciler := &ReconcileUserAccount{
-			client: client,
-			scheme: s,
-		}
+	// Register operator types with the runtime scheme.
+	s.AddKnownTypes(toolchainv1alpha1.SchemeGroupVersion, userAcc)
 
-		newUser, created, err := reconciler.ensureUser(userAcc)
-		assert.NoError(t, err)
-		assert.False(t, created)
-		assert.NotNil(t, newUser)
-		assert.Equal(t, userAcc.Name, newUser.Name)
-	})
-}
+	client := fake.NewFakeClient(objs...)
+	r := &ReconcileUserAccount{
+		client: client,
+		scheme: s,
+	}
 
-func TestEnsureIdentity(t *testing.T) {
-	logf.SetLogger(logf.ZapLogger(true))
-	username := "johnsmith"
-	userID := "34113f6a-2e0d-11e9-be9a-525400fb443d"
-	s := scheme.Scheme
-	err := apis.AddToScheme(s)
-	require.NoError(t, err)
+	req := newReconcileRequest(username)
+	identity := &userv1.Identity{}
+	user := &userv1.User{}
 
-	t.Run("create_identity", func(t *testing.T) {
-		userAcc := newUserAccount(username, userID)
-		client := fake.NewFakeClient()
-		reconciler := &ReconcileUserAccount{
+	t.Run("deleted_account_ignored", func(t *testing.T) {
+		// given
+		// No user account exists
+		objs := []runtime.Object{}
+		client := fake.NewFakeClient(objs...)
+		r := &ReconcileUserAccount{
 			client: client,
 			scheme: s,
 		}
-
-		newIdentity, created, err := reconciler.ensureIdentity(userAcc)
-		assert.NoError(t, err)
-		assert.True(t, created)
-		assert.NotNil(t, newIdentity)
+		//when
+		res, err := r.Reconcile(req)
+
+		//then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{}, res)
+
+		// Check the user is not created
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.Error(t, err)
+		assert.True(t, errors.IsNotFound(err))
+
+		// Check the identity is not created
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: getIdentityName(userAcc)}, identity)
+		require.Error(t, err)
+		assert.True(t, errors.IsNotFound(err))
 	})
 
-	t.Run("get_identity", func(t *testing.T) {
-		userAcc := newUserAccount(username, userID)
-		identity := newIdentity(userAcc)
-		client := fake.NewFakeClient(identity)
-		reconciler := &ReconcileUserAccount{
-			client: client,
-			scheme: s,
-		}
+	t.Run("create_user", func(t *testing.T) {
+		//when
+		res, err := r.Reconcile(req)
+
+		//then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{}, res)
+
+		// Check that the user account status has been updated
+		updatedAcc := &toolchainv1alpha1.UserAccount{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: req.Namespace, Name: userAcc.Name}, updatedAcc)
+		require.NoError(t, err)
+		assert.Equal(t, toolchainv1alpha1.StatusProvisioning, updatedAcc.Status.Status)
```

Comment:
I think that the `Status.Error` field should be verified as well

Reply:
Addressed in https://github.com/codeready-toolchain/member-operator/pull/25/commits/81f9d57c0d73f11d0430e1ae57b28891038d1a96